docs/dev/metrics.md

```diff
@@ -0,0 +1,213 @@
+`mapi_machine_set_status_replicas`, `mapi_machine_set_status_replicas_available`,
+and `mapi_machine_set_status_replicas_ready` entries. These individual metric
+entries help to provide current information about the state of each MachineSet.
+
+```
+# HELP mapi_machineset_items Count of machinesets at the apiserver
+# TYPE mapi_machineset_items gauge
+mapi_machineset_items 3
+# HELP mapi_machine_set_status_replicas Information of the mapi managed Machineset's status for replicas
+# TYPE mapi_machine_set_status_replicas gauge
+mapi_machine_set_status_replicas{name="ocp-cluster-rndpg-worker-us-east-2a",namespace="openshift-machine-api"} 1
+mapi_machine_set_status_replicas{name="ocp-cluster-rndpg-worker-us-east-2b",namespace="openshift-machine-api"} 1
+mapi_machine_set_status_replicas{name="ocp-cluster-rndpg-worker-us-east-2c",namespace="openshift-machine-api"} 1
+# HELP mapi_machine_set_status_replicas_available Information of the mapi managed Machineset's status for available replicas
+# TYPE mapi_machine_set_status_replicas_available gauge
+mapi_machine_set_status_replicas_available{name="ocp-cluster-rndpg-worker-us-east-2a",namespace="openshift-machine-api"} 1
+mapi_machine_set_status_replicas_available{name="ocp-cluster-rndpg-worker-us-east-2b",namespace="openshift-machine-api"} 1
+mapi_machine_set_status_replicas_available{name="ocp-cluster-rndpg-worker-us-east-2c",namespace="openshift-machine-api"} 1
+# HELP mapi_machine_set_status_replicas_ready Information of the mapi managed Machineset's status for ready replicas
+# TYPE mapi_machine_set_status_replicas_ready gauge
+mapi_machine_set_status_replicas_ready{name="ocp-cluster-rndpg-worker-us-east-2a",namespace="openshift-machine-api"} 1
+mapi_machine_set_status_replicas_ready{name="ocp-cluster-rndpg-worker-us-east-2b",namespace="openshift-machine-api"} 1
+mapi_machine_set_status_replicas_ready{name="ocp-cluster-rndpg-worker-us-east-2c",namespace="openshift-machine-api"} 1
+# HELP mapi_machineset_created_timestamp_seconds Timestamp of the mapi managed Machineset creation time
+# TYPE mapi_machineset_created_timestamp_seconds gauge
+mapi_machineset_created_timestamp_seconds{api_version="machine.openshift.io/v1beta1",name="ocp-cluster-rndpg-worker-us-east-2a",namespace="openshift-machine-api"} 1.589550153e+09
+mapi_machineset_created_timestamp_seconds{api_version="machine.openshift.io/v1beta1",name="ocp-cluster-rndpg-worker-us-east-2b",namespace="openshift-machine-api"} 1.589550153e+09
+mapi_machineset_created_timestamp_seconds{api_version="machine.openshift.io/v1beta1",name="ocp-cluster-rndpg-worker-us-east-2c",namespace="openshift-machine-api"} 1.589550153e+09
+```
+
+## Metrics about the internal state of the operator
+
+These metrics show various details about the running state of the operator.
+They can be used to diagnose issues such as increased memory or cpu usage and
+other system resource related queries.
+
+```
```

Comment:
could you point me to where are these coming from?

Reply:
i found these by scraping the running pod, but i will dig into the code a little to figure out what is generating them.

Reply:
so, apparently these are coming from the prometheus go-client. i am trying to find some documentation that i could link to, i think it will make these cleaner.